process/fuser.rs

```diff
@@ -0,0 +1,1107 @@
+        read_proc_mounts(mount_list)?;
+        *device_list = DeviceList::new(names.clone(), st.st_dev);
+        *need_check_map = true;
+    } else {
+        let st = stat(&names.filename.to_string_lossy())?;
+        *inode_list = InodeList::new(names.clone(), st.st_dev, st.st_ino);
+    }
+    Ok(())
+}
+
+fn handle_tcp_namespace(
+    names: &mut Names,
+    tcp_connection_list: &mut IpConnections,
+    inode_list: &mut InodeList,
+    net_dev: u64,
+) -> Result<(), std::io::Error> {
+    let tcp_connection_list = parse_inet(names, tcp_connection_list).unwrap();
+    *inode_list = find_net_sockets(inode_list, &tcp_connection_list, "tcp", net_dev)?;
+    Ok(())
+}
+
+fn handle_udp_namespace(
+    names: &mut Names,
+    udp_connection_list: &mut IpConnections,
+    inode_list: &mut InodeList,
+    net_dev: u64,
+) -> Result<(), std::io::Error> {
+    let udp_connection_list = parse_inet(names, udp_connection_list).unwrap();
+    *inode_list = find_net_sockets(inode_list, &udp_connection_list, "udp", net_dev)?;
+    Ok(())
+}
+
+fn print_matches(name: &mut Names, user: bool) -> Result<(), io::Error> {
+    let mut proc_map: BTreeMap<i32, (String, u32)> = BTreeMap::new();
+    let mut name_has_procs = false;
+    let mut len = name.filename.to_string_lossy().len() + 1;
+
+    eprint!("{}:", name.filename.display());
+    while len < NAME_FIELD {
+        len += 1;
+        eprint!(" ");
+    }
+    io::stderr().flush()?;
+
+    for procs in name.matched_procs.iter() {
+        if procs.proc_type == ProcType::Normal {
+            name_has_procs = true;
+        }
+
+        let entry = proc_map
+            .entry(procs.pid)
+            .or_insert((String::new(), procs.uid));
+
+        match procs.access {
+            Access::Root => entry.0.push_str("r"),
+            Access::Cwd => entry.0.push_str("c"),
+            Access::Exe => entry.0.push_str("e"),
+            Access::Mmap => entry.0.push_str("m"),
+            _ => (),
+        }
+    }
+
+    if !name_has_procs {
+        // exit if no processes matched
+        return Ok(());
+    }
+
+    for (pid, (access, uid)) in proc_map {
+        let width = if pid.to_string().len() > 4 { " " } else { "  " };
+
+        print!("{}{}", width, pid);
+        io::stdout().flush()?;
+
+        eprint!("{}", access);
+        if user {
+            let owner = unsafe {
+                CStr::from_ptr(libc::getpwuid(uid).as_ref().unwrap().pw_name)
+                    .to_str()
+                    .unwrap()
+            };
+            eprint!("({})", owner);
+        }
+        io::stderr().flush()?;
+    }
+
+    eprint!("\n");
+    Ok(())
+}
+
+/// Scans the `/proc` directory for process information and checks various access types.
+fn scan_procs(
+    need_check_map: bool,
+    names: &mut Names,
+    inode_list: &InodeList,
+    device_list: &DeviceList,
+    unix_socket_list: &UnixSocketList,
+    net_dev: u64,
+) -> Result<(), io::Error> {
+    let my_pid = std::process::id() as i32;
```

Comment:
I suggest make scan_procs async using tokio::fs::read_dir instead of synchronous reading, which works asynchronously. to increase productivity